src/lib.rs

```diff
@@ -525,3 +525,20 @@ macro_rules! matches {
         }
     }
 }
+
+#[test]
+#[allow(deprecated)]
+fn test_type_sizes() {
```

Comment:
I'm curious if this has been tested on an x86 (32 bit) platform?

IME FreeBSD x86 is always little weird about sizings of things (different than typical Linux anyway).

If I get around to it, I'll try to compile this PR on a FreeBSD 13 VM.